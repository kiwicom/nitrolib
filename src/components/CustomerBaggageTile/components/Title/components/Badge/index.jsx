// @flow
import * as React from "react";
import Badge from "@kiwicom/orbit-components/lib/Badge";
import Money from "@kiwicom/orbit-components/lib/icons/Money";
import Alert from "@kiwicom/orbit-components/lib/icons/Alert";
import Reload from "@kiwicom/orbit-components/lib/icons/Reload";
import Tooltip from "@kiwicom/orbit-components/lib/Tooltip";

import { Consumer } from "../../../../../../services/currency/context";
import { format } from "../../../../../../records/Currency";
import type { OrderStatusType } from "../../../../../../records/Baggage";
import Translate from "../../../../../Translate/index";

const getTooltipText = (status: OrderStatusType) => {
  switch (status) {
    case "unpaid":
      return <Translate t="baggage_modal.tooltip.unpaid" />;
    case "processing":
      return <Translate t="baggage_modal.tooltip.processing" />;
    case "notAvailable":
      return <Translate t="baggage_modal.tooltip.not_available" />;
    default:
      return null;
  }
};

const getBadge = (status: OrderStatusType, price?: number) => {
  switch (status) {
    case "unpaid":
      return (
        <Badge type="warning">
          <Money size="small" />
          <Consumer>
            {({ currency }) => (
              <Translate
                t="baggage_modal.badge.unpaid"
                values={{ price: price && format(currency, price) }}
              />
            )}
          </Consumer>
        </Badge>
      );
    case "processing":
      return (
        <Badge type="info">
          <Reload size="small" />
          <Translate t="baggage_modal.badge.processing" />
        </Badge>
      );
    case "notAvailable":
      return (
        <Badge type="dark">
          <Alert size="small" />
          <Translate t="baggage_modal.badge.not_available" />
        </Badge>
      );
    default:
      return null;
  }
};

type TitleBadgeProps = {
  orderStatus: "unpaid" | "processing" | "notAvailable",
  price?: number,
};

const TitleBadge = ({ orderStatus, price }: TitleBadgeProps) => (
  <Tooltip content={getTooltipText(orderStatus)} preferredPosition="top" size="small">
    <>{getBadge(orderStatus, price)}</>
  </Tooltip>
);

export default TitleBadge;
