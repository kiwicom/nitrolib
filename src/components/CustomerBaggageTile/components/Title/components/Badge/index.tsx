import * as React from "react";
import Tooltip from "@kiwicom/orbit-components/lib/Tooltip";

import { OrderStatusType } from "../../../../../../records/Baggage";
import getTooltipText from "./services/getTooltipText";
import getBadge from "./services/getBadge";

type TitleBadgeProps = {
  orderStatus: OrderStatusType,
  price: ?number,
};

const TitleBadge = ({ orderStatus, price }: TitleBadgeProps) => (
  <Tooltip content={getTooltipText(orderStatus)} preferredPosition="top" size="small">
    <>{getBadge(orderStatus, price)}</>
  </Tooltip>
);

export default TitleBadge;
