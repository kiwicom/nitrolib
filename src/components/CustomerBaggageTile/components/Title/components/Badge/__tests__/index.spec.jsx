// @flow strict
import * as React from "react";
import { shallow } from "enzyme";

import Badge from "../index";

const props = {
  orderStatus: "unpaid",
  price: {
    currency: "EUR",
    amount: 21,
    base: 21,
    service: 0,
    serviceFlat: 0,
  },
};

describe("#Badge", () => {
  test("renders status unpaid", () => {
    const wrapper = shallow(<Badge {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders status processing", () => {
    const wrapper = shallow(<Badge orderStatus="processing" />);
    expect(wrapper).toMatchSnapshot();
  });
  test("renders status not available", () => {
    const wrapper = shallow(<Badge orderStatus="notAvailable" />);
    expect(wrapper).toMatchSnapshot();
  });
});
