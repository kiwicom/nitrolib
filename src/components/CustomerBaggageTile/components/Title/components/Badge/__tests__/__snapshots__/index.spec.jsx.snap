// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Badge renders status not available 1`] = `
<Tooltip
  inline={false}
  mobile={false}
  position="top"
  tip={
    <ForwardRef>
      <Translate
        html={false}
        t="baggage_modal.tooltip.not_available"
        transform={[Function]}
        values={Object {}}
      />
    </ForwardRef>
  }
>
  <Badge
    type="dark"
  >
    <Alert
      size="small"
    />
    <Translate
      html={false}
      t="baggage_modal.badge.not_available"
      transform={[Function]}
      values={Object {}}
    />
  </Badge>
</Tooltip>
`;

exports[`#Badge renders status processing 1`] = `
<Tooltip
  inline={false}
  mobile={false}
  position="top"
  tip={
    <ForwardRef>
      <Translate
        html={false}
        t="baggage_modal.tooltip.processing"
        transform={[Function]}
        values={Object {}}
      />
    </ForwardRef>
  }
>
  <Badge
    type="info"
  >
    <Reload
      size="small"
    />
    <Translate
      html={false}
      t="baggage_modal.badge.processing"
      transform={[Function]}
      values={Object {}}
    />
  </Badge>
</Tooltip>
`;

exports[`#Badge renders status unpaid 1`] = `
<Tooltip
  inline={false}
  mobile={false}
  position="top"
  tip={
    <ForwardRef>
      <Translate
        html={false}
        t="baggage_modal.tooltip.unpaid"
        transform={[Function]}
        values={Object {}}
      />
    </ForwardRef>
  }
>
  <Badge
    type="warning"
  >
    <Money
      size="small"
    />
    <ContextConsumer>
      <Component />
    </ContextConsumer>
  </Badge>
</Tooltip>
`;
